Add filter for post types

edmundcwm opened the pull request:
Add filter for excluding specific post types from search.

Commit:
Remove formation forms and  entries from multisite search

Commit:
Utilize generate_dynamic_placeholders() helper method for $sites_match query

rheinardkorf commented:
@edmundcwm , it would be ideal to not have the posts get indexed in the first place.  Apply a filter like `skipped_posts` here: https://github.com/xwp/multisite-search/blob/master/php/Admin/class-index.php#L32

The root cause of the problem is here: https://github.com/xwp/multisite-search/blob/master/php/Hooks/class-post-type.php#L44

Even though it defaults to `page` and `post`, this function will index ANYTHING.

rheinardkorf commented:
A seperate issue, the code in https://github.com/xwp/multisite-search/blob/master/php/CLI/class-index.php#L30 will ONLY index `post` and `page`.  This would need to be fixed to include other types. 

Recommend this as a seperate issue.

rheinardkorf commented on php/Utility/class-search.php:
```diff
@@ -61,6 +63,9 @@ public function query(
 		$caps     = $escalated_caps ? '(.*)' : $caps;
 		$keywords = str_replace( '**', ' ', $keywords );
 
+		// List of post types to exclude from search.
```
See other comments on the PR to avoid requiring this code in the first place.

Commit:
Remove logic for excluding form entries

It is more appropriate to perform the exclusion before the data gets indexed.

Hence there is no point for having the logic here.

Commit:
Remove unused parameter

Commit:
Add filter to determine what post types should be indexed when using CLI

Commit:
Add post type validation when "wp_insert_hook" is fired

This will ensure the current post is of a valid post type before we index it.

edmundcwm commented:
PR edited!
